eliot/_message.py

```diff
@@ -106,16 +125,49 @@ def write(self, logger=None, action=None):
         """
         if logger is None:
             logger = _output._DEFAULT_LOGGER
-        if action is None:
-            action = currentAction()
-        if action is None:
-            action = _defaultAction
-        contents = self._contents.copy()
-        contents["timestamp"] = self._timestamp()
-        if action is not None:
-            contents["task_uuid"] = action._identification["task_uuid"]
-            contents["task_level"] = action._nextTaskLevel().level
-        logger.write(contents, self._serializer)
+        logged = self._freeze(action=action)
+        logger.write(logged.asDict(), self._serializer)
+        return logged
+
+
+
+class LoggedMessage(namedtuple('LoggedMessage', (
+        'timestamp', 'task_uuid', 'task_level', 'contents'))):
+    """
+    A L{Message} that has been logged.
+
+    @ivar timestamp: The Unix timestamp of when the message was logged.
+    @ivar task_uuid: The UUID of the task in which the message was logged.
+    @ivar task_level: The L{TaskLevel} of this message appears within the task.
+    @ivar contents: A C{dict}, the message contents without Eliot metadata.
+    """
+
+    @classmethod
+    def fromDict(cls, loggedDictionary):
```

Comment:
Eliot is switching over to PEP8 (slowly), so I suggest using underscore names for anything new.